src/components/Hotel/RoomOptions/Room/populateRoomArray.js

```diff
@@ -0,0 +1,14 @@
+export default function populateRoomArray(isSelected, room) {
+  const roomArray = [];
+  let iconSelected = false;
+  for (let i = 0; i < room.roomVacancies; i++) {
+    if (i >= room.ocuppiedVacancies) {
+      if (isSelected && !iconSelected) {
+        roomArray.push("IoSelected");
+        iconSelected = true;
+      } else roomArray.push("IoPersonOutline");
+    } else roomArray.push("IoPerson");
+  }
```

Comment:
Nesse ponto acho que o for, poderia ser substituído por um ArrayFrom, algo como: 

```
const roomArray = Array.from({ length: room.roomVacancies } , (_, index) => {
   if (index >= room.ocuppiedVacancies) {
      if ( isSelected && !iconSelected ){
        return "IoSelected" 
        iconSelected = true
      }
   } else {
      return "IoPerson"
   }
})
```

Dai garantiríamos a uma melhor semântica no código.

Reply:
posso só utilizar o map em vez do from?